Client/Frontend/Browser/BrowserPrompts.swift

```diff
@@ -35,12 +42,33 @@ class JSPromptAlertController: UIAlertController {
 protocol JSAlertInfo {
     func alertController() -> JSPromptAlertController
     func cancel()
+    /// Informs JSPromptAlertController that the completion handler has already been called.
+    var didComplete: Bool { get }
 }
 
-struct MessageAlert: JSAlertInfo {
+class MessageAlert: JSAlertInfo {
```

Comment:
Looking at the code, I'm unsure as to why we're changing this from a `struct` to a `class`. It's early in the morning and I may be entirely missing it though.

Also applies in the other places this appears in these changes.

Reply:
Since `function completionHandler()` sets the internal _completionHandler var to nil once called (to ensure the completion handler is only ever called once), keeping it as a struct would have required each function in the struct to be declared as mutating as they can all call down to the mutating completionHandler().

As a class, we also then do not have to worry about someone keeping a copy of the struct before the completionHandler is called, and then using the completionHandler AFTER the original version of the struct has called its completion handler.